HaikuPorter/DependencyResolver.py

```diff
@@ -63,7 +63,7 @@ class DependencyResolver(object):
 	def __init__(self, buildPlatform, requiresTypes, repositories, **kwargs):
 		self._providesManager = ProvidesManager()
 		self._platform = buildPlatform
-		self._requiresTypes = requiresTypes
+		self._requiresTypes = list(requiresTypes)
```

Comment:
I'd say `requiresTypes.copy()` (or even `.deepcopy()` if needed) would make the intent clearer if it's just about making a copy? Or is the cast to a list also useful for another reason?

Reply:
yes, it's a constructor that will make a list from whatever you pass as a parameter.

If the parameter is already a list, it makes a copy of it. If it's something else, it (attempts to) convert it into a list.

Using the copy() method would preserve the original type of the object.